sys/usb/usbus/dfu/dfu.c

```diff
@@ -101,7 +101,15 @@ void usbus_dfu_init(usbus_t *usbus, usbus_dfu_device_t *handler, unsigned mode)
     DEBUG("DFU: initialization\n");
     assert(usbus);
     assert(handler);
+#if defined(FLASHPAGE_SIZE)
     assert((SLOT0_OFFSET % FLASHPAGE_SIZE) == 0);
+#elif defined(FLASHPAGE_MIN_SECTOR_SIZE)
+    /* STM32F2/4/7 MCUs use sectors instead of pages, where the minimum sector
+     * size is defined by FLASHPAGE_MIN_SECTOR_SIZE, which is 16KB or 32KB
+     * (the first sector) depending on the CPU_MODEL. In this case SLOT0_OFFSET
+     * must be a multiple of the minimum sector size to cover a whole sector. */
+    assert((SLOT0_OFFSET % FLASHPAGE_MIN_SECTOR_SIZE) == 0);
```

Comment:
```suggestion
    static_assert((SLOT0_OFFSET % FLASHPAGE_MIN_SECTOR_SIZE) == 0);
```

Reply:
May I squash it directly to avoid an extra CI round? The changes are pretty small.

Reply:
Of course!